pandas/core/index.py

```diff
@@ -391,6 +395,11 @@ def _shallow_copy(self, values=None, infer=False, **kwargs):
 
         if infer:
             attributes['copy'] = False
+            if self._infer_as_myclass:
```

Comment:
@sinhrks not questing the need for this! just whether we need ``infer`` AND ``_infer_as_myclass`` or could we just use a single class variable for this (rather than the passed kw ``infer``). Just confusing to have 2 sort-of-similar ways of doing this.